src/components/NavButton.tsx

```diff
@@ -6,14 +6,18 @@ export interface NavButtonProps {
     label: string,
     targetUrl: string,
     icon: ReactNode
+    setNavOpened?: React.Dispatch<React.SetStateAction<boolean>>
 }
 
-const NavButton: FC<NavButtonProps> = ({ label, targetUrl, icon }) => {
+const NavButton: FC<NavButtonProps> = ({ label, targetUrl, icon, setNavOpened }) => {
     const navigate = useNavigate()
     const location = useLocation()
     const isActive = location.pathname == targetUrl
     return (
-        <Button onClick={() => { navigate(targetUrl) }}
+        <Button onClick={() => { 
+            navigate(targetUrl)
+            setNavOpened && setNavOpened(false)
```

Comment:
Since setNavOpened is currently optional, the check here is to prevent the TS compiler for complaining that setNavOpened could be undefined. By making it a required value, the operational check will not be needed here.